src/aura/AuraAvatarMultiToken.sol

```diff
@@ -136,20 +133,14 @@ contract AuraAvatarTwoToken is BaseAvatar, PausableUpgradeable, AuraAvatarUtils,
     ////////////////////////////////////////////////////////////////////////////
 
     /// @notice Derives the assets to be handled and corresponding reward pools based on the Aura Pool IDs supplied.
-    /// @param _pid1 Pool ID of the first token.
-    /// @param _pid2 Pool ID of the second token.
-    constructor(uint256 _pid1, uint256 _pid2) {
-        pid1 = _pid1;
-        pid2 = _pid2;
-
-        (address lpToken1,,, address crvRewards1,,) = AURA_BOOSTER.poolInfo(_pid1);
-        (address lpToken2,,, address crvRewards2,,) = AURA_BOOSTER.poolInfo(_pid2);
-
-        asset1 = IERC20MetadataUpgradeable(lpToken1);
-        asset2 = IERC20MetadataUpgradeable(lpToken2);
-
-        baseRewardPool1 = IBaseRewardPool(crvRewards1);
-        baseRewardPool2 = IBaseRewardPool(crvRewards2);
+    /// @param _pids Pool ID of tokens involved in the avatar
+    constructor(uint256[] memory _pids) {
+        for (uint256 i = 0; i < _pids.length; i++) {
+            pids.add(_pids[i]);
+            (address lpToken,,, address crvRewards,,) = AURA_BOOSTER.poolInfo(_pids[i]);
+            assets.add(lpToken);
+            baseRewardPools.add(crvRewards);
+        }
```

Comment:
Remove `constructor` and move to `initialize`. State variables can't be initialized in constructors for proxy contracts

Reply:
moved to the *initialize*  in 1bf3520289ab8590692162543359a3039d54602b